### ISSUE TITLE
Random Generated Text

### ISSUE THREAD

**Issue by j0nr** [heinrichheine/dokuwiki-imagereference#3]
When I use the imageref, the generated image has the text "Array" in front of it. 

Also images are right aligned by default.

I pulled Dokuwiki from Github on 2012-08-28

Here is the syntax I used for an example:

```
== Testing Imagereference ==

Look at the picture in <imgref image1>.

<imgcaption image1| Figure 1>{{:wiki:dokuwiki-128.png?nolink |Adds an ImageCaption tag}}</imgcaption>
```

Here is a preview of what I get back:

(if image doesn't work here is link: [Image](https://plus.google.com/photos/113986254461554402739/albums/5782437478862144241?authkey=CPmNz9_386DsdQ)

![imageref error](https://lh4.googleusercontent.com/-QgJ-hIHPvgU/UD9YQY64wDI/AAAAAAAABxw/pYHUmvJJMWQ/s735/dokuwiki_imageref_error.jpg)

**Comment by heinrichheine** [comment]
Well, the problem is that you putt the imgref tag before the according imgcaption tag.
Let me check if i can fix this

**Comment by j0nr** [comment]
Moving the <imgref> after the <imgcaption> fixes the "figure##" but I still get the text "Array" printing next to the image.

Cheers.

Edit. I am still getting figure##  might be a caching issue...?

**Comment by heinrichheine** [comment]
yes, the figure## is a known caching behaviour. Just clean the cache and you should see the right figure 1 etc

**Comment by heinrichheine** [comment]
ok, there is now a red warning message in the text when you put the imgref tag before the referemced imgcaption tag, or when there is no imgcaption tag. Hope that fixes the problem

**Comment by j0nr** [comment]
Yup I see the red text working nice. Is this behaviour fixed? Do you always have to put the ref after the image? 

Any joy on why I am getting the "Array" text?

Thanks

**Comment by heinrichheine** [comment]
Well, i didn't see the Array text in my installation. 
I will give it a second try this evening.

**Comment by heinrichheine** [comment]
Checked it with my dokuwiki installation, there is no Array text in front of the figure. 
Are there any other plugins you are using?

**Comment by j0nr** [comment]
Here is what plugins I have currently installed.

![Dokuwiki Installed Plugins](https://lh3.googleusercontent.com/-j0Ghqqia7oc/UEhJByBIxXI/AAAAAAAABzE/c3qBwhQHX0E/s690/dokuwiki_installed_plugins.jpg)

**Comment by patrikarlos** [comment]
Having the same issue. (Figure draw, Array; caption)... Hoping for a fix to be 'found'

**Comment by chrispx** [comment]
First of all, Thank you for updating the plugin!

I get the same behavior, so a the word "Array" is seen in front of the image. The rest of the wiki page is then totally screwed up. Hope for a fix soon. Here my plugins...

https://skitch.com/e-trinix/e1fa2/dw-plugins

Also got the following error

Warning: strlen() expects parameter 1 to be string, array given in /Users/chrisp/Sites/dokuwiki/lib/plugins/imagereference/syntax/imgcaption.php on line 194

Best
Christian

**Comment by heinrichheine** [comment]
Wow, seems to be a majer issue =)
I will give it a try. But guys, i am quite busy in the next time, so do not expect any progress until mid of october.

As hint for myself the link to the post to fix this issue https://forum.dokuwiki.org/thread/8772

**Comment by Klap-in** [comment]
Fixed by last release, see https://www.dokuwiki.org/plugin:imagereference or https://github.com/Klap-in/dokuwiki-plugin-imagereference
